src/cloudai/_core/base_runner.py

```diff
@@ -136,7 +136,7 @@ async def shutdown(self):
             return
         logging.info("Terminating all jobs...")
         for job in self.jobs:
-            self.kill_job(job)
+            self.system.kill(job)
```

Comment:
The kill() method is implemented in the system classes because these classes understand how to terminate jobs. A Slurm system will use scancel, while a Kubernetes system will use the Kubernetes Python client. The runner classes can also terminate jobs, but it seems more intuitive to do so through the system classes, based on my understanding.

Reply:
I see, this sounds right and logical. Yet I still miss the whole picture. Runner object implies that it can start, stop, get_info about a job. Indeed, it does that through a System. As I said in another comment, let's try to describe our objects (we have DEV.md BTW) and agree on it.